src/main/java/com/gooddata/lcm/LcmEntityFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2007-2018, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata.lcm;
+
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+import static com.gooddata.util.Validate.notEmpty;
+import static java.util.Collections.singletonList;
+
+/**
+ * Specification of the filter on {@link LcmEntities}.
+ */
+public class LcmEntityFilter {
+
+    private final MultiValueMap<String, String> params;
+
+    /**
+     * Creates new (empty) filter.
+     */
+    public LcmEntityFilter() {
+        params = new LinkedMultiValueMap<>(3);
+    }
+
+    /**
+     * Adds given data product to this filter.
+     * @param dataProduct data product id - must not be empty.
+     * @return this filter
+     */
+    public LcmEntityFilter withDataProduct(final String dataProduct) {
+        params.put("dataProduct", singletonList(notEmpty(dataProduct, "dataProduct")));
```

Comment:
What about to use constants for the key strings?

Reply:
fixed